migrate/list-convert.go

```diff
@@ -0,0 +1,181 @@
+// This file is part of MinIO DirectPV
+// Copyright (c) 2021, 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	directpv "github.com/minio/directpv/pkg/apis/directpv.min.io/v1beta1"
+	directcsi "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1beta5"
+	directcsiclient "github.com/minio/directpv/pkg/oldclient"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+)
+
+const (
+	MaxThreadCount = 200
+)
+
+func getKubeConfig() (*rest.Config, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+	kubeConfig := filepath.Join(home, ".kube", "config")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		if config, err = rest.InClusterConfig(); err != nil {
+			return nil, err
+		}
+	}
+	config.QPS = float32(MaxThreadCount / 2)
+	config.Burst = MaxThreadCount
+	return config, nil
+}
+
+func convertCRD(oldDriveList []directcsi.DirectCSIDrive) (newDriveList []directpv.DirectPVDrive, therror error) {
```

Comment:
we have to tie it to the existing installation steps, this function will be called during the installation process and the new drives will be created

We can perhaps call this migrateDriveCRD and we need another one for volume migration

Reply:
Ok, I am going to rename it `migrateDriveCRD` then. And I will also add the function for the volumes.